docs/get-clickhouse-docs.sh

```diff
@@ -27,5 +29,26 @@ else
     exit 1
     ;;
   esac
+
+  if [ -n "$2" ]; then
+    set_git_hook="$2"
+  elif [ -z "$1" ]; then
+    read -rp "Would you like to setup git hook for automatic update? (y|n): " set_git_hook
+  fi
+
+  if [ "$set_git_hook" = "y" ]; then
+    hook_command="$(pwd)/pull-clickhouse-docs-hook.sh $UPDATE_PERIOD_HOURS ||:"
+    hook_file=$(realpath "$(pwd)/../.git/hooks/post-checkout")
+    if grep -Faq "pull-clickhouse-docs-hook.sh" "$hook_file" 2>/dev/null; then
+      echo "Looks like the update hook already exists, will not add another one"
+    else
+      echo "Appending '$hook_command' to $hook_file"
+      echo "$hook_command" >> "$hook_file"
+      chmod u+x "$hook_file"  # Just in case it did not exist before append
+    fi
+  elif [ ! "$set_git_hook" = "n" ]; then
+    echo "Expected 'y' or 'n', got '$set_git_hook', will not setup git hook"
+  fi
+
   git clone "$git_url" "clickhouse-docs"
```

Comment:
We'd avoid broken state completely by placing the hook setup after the cloning clickhouse-docs

```suggestion

  if [ -n "$2" ]; then
    set_git_hook="$2"
  elif [ -z "$1" ]; then
    read -rp "Would you like to setup git hook for automatic update? (y|n): " set_git_hook
  fi

  git clone "$git_url" "clickhouse-docs"

  if [ "$set_git_hook" = "y" ]; then
    hook_command="$(pwd)/pull-clickhouse-docs-hook.sh $UPDATE_PERIOD_HOURS ||:"
    hook_file=$(realpath "$(pwd)/../.git/hooks/post-checkout")
    if grep -Faq "pull-clickhouse-docs-hook.sh" "$hook_file" 2>/dev/null; then
      echo "Looks like the update hook already exists, will not add another one"
    else
      echo "Appending '$hook_command' to $hook_file"
      echo "$hook_command" >> "$hook_file"
      chmod u+x "$hook_file"  # Just in case it did not exist before append
    fi
  elif [ ! "$set_git_hook" = "n" ]; then
    echo "Expected 'y' or 'n', got '$set_git_hook', will not setup git hook"
  fi
```

Reply:
> We'd avoid broken state completely by placing the hook setup after the cloning clickhouse-docs

It will not avoid "broken" state because there are many other ways to shoot your leg and break it:
 - Remove `clickhouse-docs` manually
 - Remove `clickhouse-docs/.git`
 - Move the whole `ClickHouse` dir to another location (the path to the script is absolute)
 - Just checkout an old branch that does not contain the script
 - ...

Moreover, nothing is "broken" after adding `||:`. On any errors it just prints the error message and everything else works as usual.

Reply:
Yes, that's true. With hook setup, it's already an order of magnitude better for documentation editing.